### PR TITLE
More improvements in plotting examples

### PR DESCRIPTION
**Pull request by KamalakerDadi**
-  Attempts to make example ```plot_demo_more_plotting.py``` looks like a notebook demonstration.

-  Simple documentation improvements in ```doc/plotting/index.rst```

### CODE REVIEW

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 1]
```diff
@@ -1,23 +1,42 @@
 """
-More nilearn plotting
-=====================
+More plotting tools from nilearn
+================================
+
+In this example, we demonstrate how to utilise plotting options from
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: utilize (or, more simply, use)

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 2]
```diff
@@ -1,23 +1,42 @@
...
+nilearn essential in visualizing brain image analysis results.
+
+We emphasize more on using parameters such as `display_mode` and `cut_coords`
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: emphasize the use of parameters

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 3]
```diff
@@ -1,23 +1,42 @@
...
+with plotting function :func:`nilearn.plotting.plot_stat_map`. Also,
+we show how to use various features such as `add_edges`, `add_contours`,
+`add_markers` essential in visualizing regions of interest images or
+mask images overlaying on subject specific anatomical/EPI image.
+The display features shown here are inherited from class
+:class:`nilearn.plotting.displays.OrthoSlicer`.
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: from the XXXXXX class

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 4]
```diff
@@ -1,23 +1,42 @@
...
+
+The parameter `display_mode` is used to draw brain slices along given
+specific directions where directions can be either of these 'ortho',
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: , where

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 5]
```diff
@@ -1,23 +1,42 @@
...
+'x', 'y', 'z', 'xy', 'xz', 'yz'. whereas parameter `cut_coords`
+is used to specify limited number of slices to visualize along given
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: a limited number of slices

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 6]
```diff
@@ -26,65 +45,179 @@
 localizer_cmap_filename = localizer_dataset.cmaps[1]
 
 ########################################
+# Now, we show from here how to visualize the retrieved datasets using plotting
+# tools from nilearn.
+
+from nilearn import plotting
+
+########################################
+# Visualizing contrast map in three different orthogonal views - 'saggital',
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: sagittal

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 7]
```diff
@@ -26,65 +45,179 @@
...
+# 'coronal' and 'axial' with coordinate positions in each view are given
+# of interest manually also with colorbar on the right side of the plots.
+
+# First argument is path to filename of the constrast map, optional argument
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: The first argument is a path to the filename of a contrast map

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 8]
```diff
@@ -26,65 +45,179 @@
...
+# display_mode is given as string 'ortho' to visualize the map in three
+# specific directions xyz and optional argument cut_coords seen as integers
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: and the optional `cut_coords` argument, that is here a list of integers,

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 9]
```diff
@@ -26,65 +45,179 @@
...
+# given in a list denotes coordinates of each slice in the order [x, y, z].
+# By default colorbar is True in plot_stat_map function without explicitly
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: By default the `colorbar` argument is set to True in plot_stat_map.

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 10]
```diff
@@ -26,65 +45,179 @@
...
 plotting.plot_stat_map(localizer_cmap_filename, display_mode='ortho',
-                       cut_coords=(36, -27, 60),
-                       title="display_mode='ortho', cut_coords=(36, -27, 60)")
+                       cut_coords=[36, -27, 60],
+                       title="display_mode='ortho', cut_coords=[36, -27, 60]")
 
 ########################################
+# Visualizing contrast map in single view 'axial' with maximum number of
+# slices in this view are limited to 5. The coordinates to cut the slices
+# are located by itself.
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: are selected automatically

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 11]
```diff
@@ -26,65 +45,179 @@
...
+
+# In this type of visualization, argument display_mode is given as string 'z'
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: the `display_mode` argument

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 12]
```diff
@@ -26,65 +45,179 @@
...
+# for axial direction and cut_coords as integer 5 without a list implies that
+# number of cuts in the slices should be maximum of 5.
 plotting.plot_stat_map(localizer_cmap_filename, display_mode='z', cut_coords=5,
                        title="display_mode='z', cut_coords=5")
 
 ########################################
+# Visualizing contrast map in another single view 'saggital' and also showing
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: sagittal

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 13]
```diff
@@ -26,65 +45,179 @@
...
+# how to select two slices of particular interest manually by giving the
+# coordinates to cut each slice.
+
+# In this type, display_mode should be given as string 'x' for saggital view
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: sagittal

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 14]
```diff
@@ -26,65 +45,179 @@
...
+# Now contrast map is shown without a colorbar on the right side.
+
+# argument colorbar should be given as False to show plots without a colorbar on
+# the right side.
 plotting.plot_stat_map(localizer_cmap_filename, display_mode='z',
                        cut_coords=1, colorbar=False,
                        title="display_mode='z', cut_coords=1, colorbar=False")
 
 ########################################
+# Now we visualize contrast map with two views - 'saggital' and 'axial' and
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: the contrast map

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 15]
```diff
@@ -26,65 +45,179 @@
...
+# coordinates are given manually to select particular cuts in two views.
+
+# argument display_mode='xz' where 'x' for saggital and 'z' for axial view.
+# argument cut_coords should match with input number of views therefore two
+# integers should be given in a list to cut slices with coordinates
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: to select the slices to be displayed

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 16]
```diff
@@ -26,65 +45,179 @@
...
 plotting.plot_stat_map(localizer_cmap_filename, display_mode='xz',
-                       cut_coords=(36, 60),
-                       title="display_mode='xz', cut_coords=(36, 60)")
+                       cut_coords=[36, 60],
+                       title="display_mode='xz', cut_coords=[36, 60]")
 
 ########################################
+# Visualizing contrast map with 'coronal', 'saggital' views and coordinates are
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: the contrast map

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 17]
```diff
@@ -26,65 +45,179 @@
...
-                       cut_coords=(-27, 60),
-                       title="display_mode='yz', cut_coords=(-27, 60)")
+                       cut_coords=[-27, 60],
+                       title="display_mode='yz', cut_coords=[-27, 60]")
 
 ###############################################################################
-# demo display objects with add_* methods
+# In second part, we switch to demonstrating various features add_* from
+# nilearn where each specific feature will be helpful in projecting brain
+# imaging results for user specific interpretation.
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: for further interpretation

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 18]
```diff
@@ -26,65 +45,179 @@
...
+# Import image processing tool for basic processing of functional brain image
+from nilearn import image
+
+# Compute voxel-wise mean functional image across time dimension. Now we have
+# functional image in 3D assigned in mean_haxby_img
 mean_haxby_img = image.mean_img(haxby_func_filename)
 
-# Plot T1 outline on top of the mean EPI (useful for checking coregistration)
+########################################
+# Now lets see how to use `add_edges`, method useful for checking coregistration
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: let us

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 19]
```diff
@@ -26,65 +45,179 @@
...
+# by overlaying anatomical image as edges (red) on top of mean functional image
+# (background), both being of same subject.
+
+# First, we call plotting function `plot_anat` into variable display with first
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: we call the `plot_anat` plotting function, with a background image as first argument, in this case the mean fMRI image.

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 20]
```diff
@@ -26,65 +45,179 @@
...
+# displayed as red 'r', to choose different colors green 'g' and  blue 'b'.
 display.add_edges(haxby_anat_filename)
 
 ########################################
-# Plotting outline of the mask on top of the EPI
+# Plotting outline of the mask (red) on top of the mean EPI image with
+# `add_contours`. This method is useful for region specific interpretation
+# of brain images
+
+# As seen before, we call function `plot_anat` with first argument being the
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: we call the `plot_anat` function with a background image as first argument,

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 21]
```diff
@@ -26,65 +45,179 @@
...
+# mean of functional image (background image) and argument cut_coords
+# as list for manual cut with coordinates pointing at masked brain regions
 display = plotting.plot_anat(mean_haxby_img, title="add_contours",
-                             cut_coords=(28, -34, -22))
+                             cut_coords=[28, -34, -22])
+# Now use add_contours in display object with first argument should be
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: with the path to a mask image from the Haxby dataset as first argument

File: examples/01_plotting/plot_demo_more_plotting.py [hunk 22]
```diff
@@ -26,65 +45,179 @@
...
+# argument marker_color denotes color of the sphere in this case yellow 'y'
+# and third argument marker_size denotes size of the sphere
+coords = [(28, -34, -22)]
+display.add_markers(coords, marker_color='y', marker_size=100)
+
 ###############################################################################
-# demo saving plots to file
+# Finally, saving the plots to file with two different ways
 
+# contrast map plotted with function `plot_stat_map` can be saved using an
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_more_plotting.py]: Contrast maps

**Comment by bthirion** [comment]
Great contribution, thx. I have just outlined minor wording issues.

**Comment by KamalakerDadi** [comment]
@bthirion Thanks for reviewing. I addressed your comments.

**Comment by KamalakerDadi** [comment]
Can I have one more review on this ?

**Comment by AlexandreAbraham** [comment]
Great improvements!

**Pull request merged**